Mathlib/RingTheory/LocalRing/MaximalIdeal/Basic.lean

```diff
@@ -99,3 +99,40 @@ end LocalRing
 
 theorem LocalRing.maximalIdeal_eq_bot {R : Type*} [Field R] : LocalRing.maximalIdeal R = ⊥ :=
   LocalRing.isField_iff_maximalIdeal_eq.mp (Field.toIsField R)
+
+section Nilrad_max_localization
+
+open Ideal
+
+variable {R : Type*} [CommSemiring R] {S : Type*} [CommSemiring S] [Algebra R S] {M : Submonoid R}
+
+/--
+An element `x` of a commutative local semiring is not contained in the maximal ideal
+iff it is a unit.
+-/
+theorem LocalRing.not_mem_maximalIdeal [LocalRing R] (x : R) :
```

Comment:
I think maybe this theorem should be moved to immediately follow the theorem `mem_maximalIdeal`.